### PR TITLE
AMQNET-722 Connection timing new options

### PR DESCRIPTION
**Pull request by lukeabsent**

### CODE REVIEW

**Comment by michaelandrepearce** [comment]
rebase instead of merge, to avoid merge commits in your local fork/branch.

File: docs/configuration.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ The options apply to the behavior of the NMS objects such as Connection, Session
 - **nms.requestTimeout** Timeout value that controls how long the client waits on completion of various synchronous interactions, such as opening a producer or consumer, before returning an error. Does not affect synchronous message sends. By default the client will wait indefinitely for a request to complete.
 - **nms.clientIdPrefix** Optional prefix value that is used for generated Client ID values when a new Connection is created for the JMS ConnectionFactory. The default prefix is 'ID:'.
 - **nms.connectionIdPrefix** Optional prefix value that is used for generated Connection ID values when a new Connection is created for the JMS ConnectionFactory. This connection ID is used when logging some information from the JMS Connection object so a configurable prefix can make breadcrumbing the logs easier. The default prefix is 'ID:'.
+- **nms.maxNewConnectionRatePerSec** Allowed approximated rate for how fast connection factory is allowed to create new connection. If there is more request, they will have to wait. Default value is 0 which means unlimited.
```
Comment (michaelandrepearce) [inline docs/configuration.md:26]: typically -1 means to disable/no limit, can we align please.

Comment (michaelandrepearce) [reply docs/configuration.md:26]: e.g. - **failover.maxReconnectAttempts** The number of reconnection attempts allowed before reporting the connection as failed to the client. The default is no limit or (-1).

Comment (lukeabsent) [reply docs/configuration.md]: ok, makes sense

**Pull request merged**